src/hotspot/share/interpreter/bytecodes.cpp

```diff
@@ -385,12 +385,13 @@ int Bytecodes::special_length_at(Bytecodes::Code code, address bcp, address end)
       if (end != nullptr && aligned_bcp + 3*jintSize >= end) {
         return -1; // don't read past end of code buffer
       }
-      int lo = Bytes::get_Java_u4(aligned_bcp + 1*jintSize);
-      int hi = Bytes::get_Java_u4(aligned_bcp + 2*jintSize);
-      int len = (int)(aligned_bcp - bcp) + (3 + hi - lo + 1)*jintSize;
+      // Promote calculation to 64 bits to do range checks, used by the verifier.
+      int64_t lo = (int)Bytes::get_Java_u4(aligned_bcp + 1*jintSize);
+      int64_t hi = (int)Bytes::get_Java_u4(aligned_bcp + 2*jintSize);
```

Comment:
Are the int casts needed?

Reply:
Yes they are.    get_Java_u4 returns u4 (unsigned) and the 'int' cast sign extends it to int, then further sign extended to int64_t for the length calculation.  The tableswitch bytecode operands are signed int.

```
static inline u4   get_Java_u4(address p)           { return get_Java<u4>(p); }
```

Reply:
I think this should really be a new API with a name like get_Java_s4(), but that can be a separate RFE.  There are other functions in globalDefinitions.hpp where we try really hard not to use undefined or implemention-defined behavior, and casting from unsigned to signed seems to be one of those, at least until c++20 fixes it.